Dell/recovery_backend.py

```diff
@@ -218,6 +220,45 @@ def _check_polkit_privilege(self, sender, conn, privilege):
                     sender, conn, pid, privilege, str(details))
             raise PermissionDeniedByPolicy(privilege)
 
+    def _backup_iso_efi_to_file(self, base_iso):
+        '''backing the efi partition from the base iso'''
+
+        no_options = GLib.Variant('a{sv}', {})
+        efi_guid = 'c12a7328-f81f-11d2-ba4b-00a0c93ec93b'
+        image_fd = os.open(base_iso, os.O_RDWR)
+        udisks = UDisks.Client.new_sync(None)
+        manager = udisks.get_manager()
+
+        (obj_path, out_fd_list) = manager.call_loop_setup_sync(
+                    GLib.Variant('h', 0),
+                    GLib.Variant('a{sv}', []),
+                    Gio.UnixFDList.new_from_array([image_fd]),
+                    None)
+        if obj_path == None:
+            return logging.warning("Failed to create loop device")
+
+        udisks = UDisks.Client.new_sync(None)
+        obj = udisks.get_object(obj_path)
+        if obj == None:
+            return logging.warning("Failed to get loop object")
+
+        for partition in obj.get_partition_table().get_property('partitions'):
+            if udisks.get_object(partition).get_partition().get_property('type').lower() == efi_guid:
+                filepath = udisks.get_object(partition).get_block().get_property('device')
+                print("filepath: %s" % filepath)
+
+                with open(filepath, "rb") as fr:
+                    byte = fr.read()
+                    fr.close()
+
+                with open('/tmp/backingfile-efi.img', "wb") as fw:
```

Comment:
please don't hardcode a path.  This would be a security vulnerability if another process squatted on the filename.  Instead generate a tmpfile.

Reply:
the hardcoded path has been replaced by given from tempfile `assembly_tmp` that is created at run time dynamically.
```
        file_efimage1 = os.path.join(assembly_tmp, 'boot', 'grub', 'efi.img')
        file_efimage2 = os.path.join(assembly_tmp, 'boot', 'efi.img')
```